distributed/shuffle/tests/test_shuffle.py

```diff
@@ -439,6 +442,50 @@ async def test_restarting_during_transfer_raises_killed_worker(c, s, a, b):
     await check_scheduler_cleanup(s)
 
 
+@gen_cluster(
+    client=True,
+    nthreads=[("", 1), ("", 1)],
+    config={"distributed.scheduler.allowed-failures": 0},
+)
+async def test_erred_task_before_p2p_does_not_log_event(c, s, a, b):
+    def block_and_fail_eventually(df, lock, event):
+        with worker_client() as c:
+            with lock:
+                variable = Variable("allowed_tasks", client=c)
+                allowed_tasks = variable.get()
+                if allowed_tasks > 0:
+                    variable.set(allowed_tasks - 1)
+                    return df
+                event.wait()
+                raise RuntimeError("test error")
+
+    df = dask.datasets.timeseries(
+        start="2000-01-01",
+        end="2000-02-01",
+        dtypes={"x": float, "y": float},
+        freq="10 s",
+    )
+    lock = Lock()
+    variable = Variable("allowed_tasks", client=c)
```

Comment:
FYI You could also do smth like

```suggestion
    semaphore = Semaphore(max_leases=s.total_nthreads * 2 + 1)
    
    def block_and_fail_eventually(...):
        try:
            sem.acquire(timeout=0)
        except TimeoutError:
            raise RuntimeError()
```

no strong preferences on my end.